### ISSUE TITLE
Invalid metadata on multi search with x-typesense-api-key

### ISSUE THREAD

**Issue by tudor2004** [typesense/typesense#1490]
## Description

`Invalid metadata` error thrown when passing the `x-typesense-api-key` on a multi-search request inside the body, under Typsense > v0.25.0. 

## Steps to reproduce

Start a local typesense instance with any version higher than 0.25.0

```
docker run -p 8108:8108 \
            -v$(pwd)/typesense-data:/data typesense/typesense:0.25.2 \
            --data-dir /data \
            --api-key=$TYPESENSE_API_KEY \
            --enable-cors
```

Create a basic collection and than perform a `multi_search` query on that collection like this 

```
curl --location 'http://localhost:8108/multi_search?query_by=title&q=' \
--header 'Content-Type: application/json' \
--data '{
    "searches": [
        {
            "collection": "articles",            
            "x-typesense-api-key": "xyz"
        }
        
    ]
}'

Response:
{
    "message": "Invalid metadata"
}
```

The query above, works only when the `x-typesense-api-key` is passed as an query param:

```
curl --location 'http://localhost:8108/multi_search?query_by=title&q=&x-typesense-api-key=xyz' \
--header 'Content-Type: application/json' \
--data '{
    "searches": [
        {
            "collection": "articles",            
            "x-typesense-api-key": "xyz"
        }
        
    ]
}'

Response:
{
    "results": [
        {
            "facet_counts": [],
            "found": 0,
            "hits": [],
            "out_of": 0,
            "page": 1,
            "request_params": {
                "collection_name": "articles",
                "per_page": 10,
                "q": ""
            },
            "search_cutoff": false,
            "search_time_ms": 7
        }
    ]
}
```

## Expected Behavior

In versions prior to 0.25.x the we were able to pass the `x-typesense-api-key` inside the body for each search on a `multi_search` request like[ specified in the documentation](https://typesense.org/docs/0.24.0/api/federated-multi-search.html#multi-search-parameters).

## Actual Behavior

After upgrading to 0.25.2 passing the `x-typesense-api-key` inside the body, like in the example above, is throwing an "Invalid metadata" error. 

I might be wrong, but I think this check was introduced through this PR https://github.com/typesense/typesense/pull/846 
by @kishorenc 

Strangely we cannot reproduce the same behaviour using the Typesense Cloud clusters on the same version, only when using our own cluster where we install Typesense via docker. 

## Metadata

**Typesense Version**: > 0.25.0

**OS**: Mac

**Comment by kishorenc** [comment]
Thank you for reporting this regression. Will share a fixed build next in a few days.

**Comment by kishorenc** [comment]
Fixed in `0.26.0.rc48`.

**Comment by jasonbosco** [comment]
Released in v26
